daemon/gamemode.c

```diff
@@ -604,6 +605,78 @@ GameModeContext *game_mode_context_instance()
 	return &instance;
 }
 
+/**
+ * Attempt to resolve the exe for wine-preloader.
+ * This function is used if game_mode_context_find_exe() identified the
+ * process as wine-preloader.
+ */
+static char *game_mode_resolve_wine_preloader(pid_t pid)
+{
+	static char proc_path[PATH_MAX] = { 0 };
```

Comment:
Any reason for static here (and later on)? Makes this function non-thread-safe. Stack size could be a concern but I don't think 2 4K buffers is too much issue given that default stack sizes should be significantly larger.

Reply:
I think I copied that from somewhere else and wondered about thread-safety, too. I'll change that. What about using `malloc()` to not pullute the stack? I think an allocation is not a performance problem here, this only loads at process startup time. Or I try to move this logic up into the config parser.

Reply:
@aejsmith I actually copied it from `game_mode_find_exe()` which has this static line. Since my function is always only called from that said function, it should have the same semantics, shouldn't it? But I can fix that function, too, if thread-safety is a concern here. Are these functions really called from different threads?

Reply:
I rebased a fix you can cherry-pick from.

Reply:
I don't think it can be used by multiple threads at the same time right now, just don't think it's a good idea to unnecessarily make the restriction that it can't be.

Hadn't noticed `game_mode_find_exe` had the same, so yes, that should be fixed too.

Reply:
It's actually fixed, it's one of the top commits.